File: Solutions/3005/3005.ts
function maxFrequencyElements(nums: number[]): number {
    const kMax:number = 100;
    const count:number[] = new Array(kMax + 1).fill(0);  // Initialize count array with zeros
    let ans:number = 0;

    for (const num of nums) {
        count[num]++;
    }

    const maxFreq = Math.max(...count);

    for (const freq of count) {
        if (freq === maxFreq) {
            ans += maxFreq;
        }
    }

    return ans;
};

File: Solutions/3005/3005.go

func maxFrequencyElements(nums []int) int {
	const kMax = 100
	count := make([]int, kMax+1) // Initialize count array with zeros
	var ans int

	for _, num := range nums {
		count[num]++
	}

	maxFreq := 0
	for _, freq := range count {
		if freq > maxFreq {
			maxFreq = freq
		}
	}

	for _, freq := range count {
		if freq == maxFreq {
			ans += maxFreq
		}
	}

	return ans
}